framework/source/class/qx/ui/treevirtual/celleditor/NodeEditor.js

```diff
@@ -0,0 +1,75 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2020 Patrick Buxton
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Pat Buxton (rad-pat)
+
+************************************************************************ */
+
+
+/**
+ * A cell editor factory for editing a virtualtree node label
+ */
+qx.Class.define("qx.ui.treevirtual.celleditor.NodeEditor",
+{
+  extend : qx.ui.table.celleditor.TextField,
+
+  members :
+  {
+    // overridden
+    getCellEditorValue : function(cellEditor)
+    {
+      var label = cellEditor.getValue();  // This will be the new label for the Tree Node
+
+      // validation function will be called with new and old label
+      var validationFunc = this.getValidationFunction();
+      if (validationFunc ) {
+        label = validationFunc(label, cellEditor.originalLabel);
+      }
+
+      var newValue = qx.lang.Object.clone(cellEditor.originalValue);
+      newValue.label = label;
+      return newValue;
+    },
+
+
+    // interface implementation
+    createCellEditor : function(cellInfo)
+    {
+      var cellEditor = this._createEditor();
+
+      // The value in the case of a Tree is a Node and we want the label
+      if (cellInfo.value === null || cellInfo.value === undefined) {
```

Comment:
If `cellInfo.value` is required to be an object, as indicated by the value set below, then this test might better be
```
if (cellInfo.value === null || typeof cellInfo.value != "object")
```